### PR TITLE
Expose Parse.module_type and Parse.module_expr

### PR DESCRIPTION
**Pull request by gpetiot**
These changes were made in the vendored compilerlib we use for ocamlformat, it would be nice to have it integrated upstream. This only exposes another function from `parse.ml`.

The goal of exposing this function is that we have a usecase where ocaml-lsp/merlin needs to have module types to be formatted, which will be done by ocamlformat, we used to have a dirty implementation (see https://github.com/ocaml-ppx/ocamlformat/pull/1864) in ocamlformat to circumvent not having the `Parse.module_type` function.

### CODE REVIEW

File: boot/menhir/parser.mli [hunk 1]
```diff
@@ -142,6 +142,8 @@ val parse_pattern: (Lexing.lexbuf -> token) -> Lexing.lexbuf -> (Parsetree.patte
 
 val parse_mty_longident: (Lexing.lexbuf -> token) -> Lexing.lexbuf -> (Longident.t)
 
+val parse_module_type: (Lexing.lexbuf -> token) -> Lexing.lexbuf -> (Parsetree.module_type)
```
Comment (gasche) [inline boot/menhir/parser.mli:145]: The placement of this item within the `.mli` is not great, as it sits in the middle of the `parse_*_longident` function. You could move it right above `interface` and `implementation` below.

**Review by gasche — COMMENTED** [review]
This looks very reasonable. To preserve the symmetry, could you also add a parser  for `module_expr`?

File: boot/menhir/parser.mli (hunk 1 above)
Comment (gpetiot) [reply boot/menhir/parser.mli:145]: I tried a few things but nothing impacted the order of declarations in this file.

Comment (gasche) [reply boot/menhir/parser.mli:145]: aah... I thought that I was looking at a human-written file, but this is in fact a Menhir-generated file. Sorry for the noise.

**Review by gasche — APPROVED** [review]

**Comment by gasche** [comment]
This is good to merge when the CI agrees. @gpetiot, you should add a Changes entry.

**Pull request merged**